### DISCUSSION TITLE
Encoding problem on vscode with code runner

### DISCUSSION THREAD

**Discussion by kitoutou999** [community/community#142327] [Programming Help] [answered]
### Body

Hello, I am a French developer, and I am encountering a bug on VSCode with the Code Runner extension. When I try to display the character 'é', it is replaced in the VSCode terminal by a white diamond with a question mark. Do you have any ideas about this bug? I have already set VSCode and Code Runner to UTF-8, and I don't have this issue on any other terminal or PowerShell.
![Capture d'écran 2024-10-23 005156](https://github.com/user-attachments/assets/1f437372-99d8-45c5-8069-42e7be565ede)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by buraito-kun** [comment]
In code runner command line it use a cmd.exe by default which it might not support characters outside of the basic ASCII set.

**Reply by kitoutou999** [reply]
thanks, you now if it's possible to change by an other terminal ?

**Reply by buraito-kun** [reply]
To change the command used by the Code Runner for Python files, change [the code-runner.executorMap setting](https://github.com/formulahendry/vscode-code-runner#configuration). Ex.

"code-runner.executorMap": {
    "python": "powershell -Command python",
}

ref: https://stackoverflow.com/questions/76553674/how-do-you-change-your-default-code-runner-execution-command-in-vs-code

I think with this method you can run python file using powershell as execution.

**Comment by florianppn** [comment, marked as answer]
You just need to replace per this

python 
"code-runner.executorMap": {
"python": "python -u -X utf8",
}
